packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -613,14 +623,11 @@ class InspectorController extends DisposableController
 
     final group = _selectionGroups.next;
 
-    if (firstFrame) {
-      final queryParams = loadQueryParams();
-      if (queryParams.containsKey(inspectorRefQueryParam)) {
-        await group.setSelectionInspector(
-          InspectorInstanceRef(queryParams[inspectorRefQueryParam]),
-          false,
-        );
-      }
+    if (inspectorRef != null) {
```

Comment:
What is the motivation behind moving the query param check out of `updateSelectionFromService`? `updateSelectionFromService` is also called from `onInspectorSelectionChanged` - is it safe that we are no longer checking the query params from that call site?

Reply:
It was just because I wanted to provide a different value for clicking the errors and it felt a little awkward adding a parameter to allow you to supply it, but then reading it from the querystring based on the value of a different parameter.

> is it safe that we are no longer checking the query params from that call site?

The query string code was inside an `if (firstFrame)` so it wouldn't have executed via that code path.

(happy to change this if you prefer it the other way - presumably we'd always prefer the parameter if one was provided over the querystring?)

Reply:
updateSelectionFromService is the breadcrumb for how you want to implement this efficiently. That code path has the same challenges of you need to update the location in the summary tree that is closest to the current selection and update the details selection. The code to solve it is ugly and convoluted so it is much easier to make sure you use the same code path than make another code path that does the same thing.
Basically what happens is given an inspector ref it computes the first parent of the inspector ref that is in the summary tree and selects that.
If the node is already in the detail tree it selects that node, otherwise it reloads the detail tree rooted appropriately to include the node you need from the details tree.